src/include/cell_inline.h

```diff
@@ -53,33 +53,62 @@ __cell_pack_value_validity(WT_SESSION_IMPL *session, uint8_t **pp, WT_TIME_WINDO
     ++*pp;
 
     flags = 0;
-    if (tw->start_ts != WT_TS_NONE) {
+    const bool pack_prepare_info_to_start = F_ISSET(S2BT(session), WT_BTREE_DISAGGREGATED) &&
+      tw->prepare && tw->prepare_ts != WT_TS_NONE &&
```

Comment:
May be no need to check tw->prepare_ts != WT_TS_NONE. Instead we turn that into an assert?